odoo/fields.py

```diff
@@ -3816,6 +3816,11 @@ def convert_to_record(self, value, record):
             if not property_model and 'domain' in property_definition:
                 del property_definition['domain']
 
+            if property_definition.get('type') in ('selection', 'tags'):
+                # always set at least an empty array if there's no option
+                key = property_definition['type']
+                property_definition[key] = property_definition.get(key) or []
+
```

Comment:
Nitpicking, wouldn't `setdefault` works in this case?
```suggestion
            if property_definition.get('type') in ('selection', 'tags'):
                # always set at least an empty array if there's no option
                property_definition.setdefault(property_definition['type'], [])

```

Reply:
but it won't fix it if we have existing record in database with False as value